src/bin/pgcopydb/ld_transform.c

```diff
@@ -1169,6 +1169,64 @@ parseMessage(StreamContext *privateContext, char *message, JSON_Value *json)
 }
 
 
+/* append the new insert statement if targets same table as previous */
+static bool
+coalesceInsertStatements(LogicalMessageInsert *last,
+						 LogicalMessageInsert *stmt)
+{
+	if (strcmp(last->nspname, stmt->nspname) != 0 ||
+		strcmp(last->relname, stmt->relname) != 0)
+	{
+		return false;
+	}
```

Comment:
I think we're missing a check that the list of columns in the INSERT INTO target list is the same for both the statements. We can't add a VALUES entry otherwise.